Packs/Expanse/Integrations/Expanse/Expanse.py

```diff
@@ -698,35 +714,69 @@ def fetch_incidents_command():
     page_token = None
     incidents = []
 
-    while more_events:
-        event_incidents, page_token = fetch_events_incidents_command(start_date, end_date, token, page_token)
-        for incident in event_incidents:
-            demisto.debug("Adding event incident name={name}, type={type}, severity={severity}".format(**incident))
-        incidents += event_incidents
-        if page_token is None:
-            more_events = False
-
-    # Fetch Behavior
-    if BEHAVIOR_ENABLED:
-        more_behavior = True
-        next_offset = None
-
-        while more_behavior:
-            behavior_incidents, next_offset = fetch_behavior_incidents_command(start_date, token, next_offset)
-            for incident in behavior_incidents:
-                demisto.debug("Adding behavior incident name={name}, type={type}, severity={severity}".format(**incident))
-            incidents += behavior_incidents
-            if next_offset is None:
-                more_behavior = False
-
-    # Send all Incidents
-    demisto.incidents(incidents)
-
-    # Save last_run
-    demisto.setLastRun({
-        "complete_for_today": True,
-        "start_time": yesterday
-    })
+    # Check if we've stored any events in the integration cache
+    cache = demisto.getIntegrationContext()
+    stored_incidents = cache.get("incidents")
+
+    if stored_incidents is None:
```

Comment:
it would be great if we could add a unit test for that flow to verify the cache mechanism works as expected